pokemongo_bot/event_handlers/chat_handler.py

```diff
@@ -207,73 +67,86 @@ def get_caught(self, chat_id, num, order):
         if not num.isnumeric():
             num = 10
         else:
             num = int(num)
 
         if order not in ["cp", "iv"]:
             order = "iv"
+
         with self.bot.database as conn:
             cur = conn.cursor()
             cur.execute("SELECT * FROM vanish_log ORDER BY " + order + " DESC LIMIT " + str(num))
             vanished = cur.fetchall()
-            outMsg = ''
-            if vanished:
-                for x in vanished:
-                    outMsg += '*' + x[0] + '* ' + '(_CP:_ ' + str(int(x[1])) + ') (_IV:_ ' + str(x[2]) + ')\n'
-                self.sendMessage(chat_id=chat_id, parse_mode='Markdown', text="".join(str(outMsg)))
-            else:
-                self.sendMessage(chat_id=chat_id, parse_mode='Markdown', text="No Pokemon Vanished Yet.\n")
+            return vanished
+
+    def get_event(self, event, formatted_msg, data):
+        msg = None
+        if event == 'level_up':
+            msg = "level up ({})".format(data["current_level"])
+        elif event == 'pokemon_caught':
+            trigger = None
+            if data["pokemon"] in self.pokemons:
+                trigger = self.pokemons[data["pokemon"]]
+            elif "all" in self.pokemons:
+                trigger = self.pokemons["all"]
+            if trigger:
+                if ((not "operator" in trigger or trigger["operator"] == "and") and data["cp"] >= trigger["cp"] and
+                            data["iv"] >= trigger["iv"]) or \
+                        ("operator" in trigger and trigger["operator"] == "or" and (
+                                data["cp"] >= trigger["cp"] or data["iv"] >= trigger["iv"])):
+                    msg = "Caught {} CP: {}, IV: {}".format(data["pokemon"], data["cp"], data["iv"])
+        elif event == 'egg_hatched':
+            msg = "Egg hatched with a {} CP: {}, IV: {} (A/D/S {})".format(data["name"], data["cp"], data["iv_pct"],
+                                                                           data["iv_ads"])
+        elif event == 'bot_sleep':
+            msg = "I am too tired, I will take a sleep till {}.".format(data["wake"])
+        elif event == 'catch_limit':
+            msg = "*You have reached your daily catch limit, quitting.*"
+        elif event == 'spin_limit':
+            msg = "*You have reached your daily spin limit, quitting.*"
+        else:
+            return formatted_msg
 
-        def evolve(self, chatid, uid):
-            # TODO: here comes evolve logic (later)
-            self.sendMessage(chat_id=chatid, parse_mode='HTML', text="Evolve logic not implemented yet")
-            return
+        return msg
 
-        def upgrade(self, chatid, uid):
-            # TODO: here comes upgrade logic (later)
-            self.sendMessage(chat_id=chatid, parse_mode='HTML', text="Upgrade logic not implemented yet")
-            return
\ No newline at end of file
+    def sendTeleMessage(self, chat_id=None, parse_mode='Markdown', text=None):
```

Comment:
What is this supposed to do here?

chat_handler is an abstract handler, not intended to have any telegram-specific logic. The communication to the frontend (telegram and discord currently existing) is done by telegram_handler and discord_handler.